drf_tweaks/test_utils.py

```diff
@@ -0,0 +1,64 @@
+# -*- coding: utf-8 -*-
+from django.conf import settings
+from django.db.backends.utils import CursorWrapper
+from rest_framework.test import APIClient, APITestCase
+
+import warnings
+
+test_query_counter = 0
+
+
+def hacked_execute(self, sql, params=()):
+    global test_query_counter
+    if "SAVEPOINT" not in sql:
+        test_query_counter += 1
+    return self.old_execute(sql, params)
+
+
+class query_counter(object):
+    def __enter__(self):
+        # reset counter
+        global test_query_counter
+        test_query_counter = 0
+
+        # patching CursorWrapper
+        CursorWrapper.old_execute = CursorWrapper.execute
+        CursorWrapper.execute = hacked_execute
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        # restore previous function
+        CursorWrapper.execute = CursorWrapper.old_execute
+
+        if exc_type is None:
+            global test_query_counter
+
+            if test_query_counter > getattr(settings, "TEST_QUERY_NUMBER_RAISE_ERROR", 15):
+                raise Exception("Too many queries executed: %d" % test_query_counter)
```

Comment:
Could you make some TooManySQLQueriesException(Exception) and change it here and in test below?